es-core/src/resources/Font.cpp

```diff
@@ -610,42 +610,69 @@ float Font::getLetterHeight()
 
 //the worst algorithm ever written
 //breaks up a normal string with newlines to make it fit xLen
+// supported UTF-8 text (wordwrap and cheap hyphenation)
 std::string Font::wrapText(std::string text, float xLen)
 {
 	std::string out;
+	std::string line, word;
+
+	size_t pos = 0, pos2 = 0;
+	while (0 < text.length()) {
+		size_t n = pos;
+		for (; n < (int)text.length(); n++) {
+			if (pos == 0 && xLen < sizeText(text.substr(pos, n)).x()) {
+				n--;
+				break;
+			}
+			char c = text[n];
+			if ('0' <= c && c <= '9') continue;
```

Comment:
So let me try to paraphrase this new strategy; 
1. in stead of searching for a white-space character (space, \t, \n), we now try until we arrive at an out-of-range character, 
2. which might happen to be a whitespace, but might also be a non-ascii character.
3. If so, check if we were able to make -any- progress since `pos`
4. if we made some progress, split at this point.
5. if not (pos == n), get next characterlocation, this time taking into account utf-8 encoding.
6. if we find a `\n` next, cut here.
7. otherwise, continue until it no longer fits, then cut.

To get the actual next-character-in-utf-8 you use `getNextCursor()`, which returns the stepsize needed to progress to the next character location. Note that this also works fine for ascii texts, (it will return 1).
Actually, what you should (i.m.o.) use is `readUnicodeChar()`, to get the next character from the next location, and auto-advance the cursor. 
I think I have a working prototype, but I'd like you to take a look at it as well.
I've send[ a PR your way for you to test](https://github.com/eagle0wl/EmulationStation/pull/1) this.